db/bigtable_eth1.go

```diff
@@ -3273,9 +3297,16 @@ func (bigtable *Bigtable) GetMetadataForAddress(address []byte) (*types.Eth1Addr
 
 					return nil
 				})
-			}
 
-			if column.Column == ACCOUNT_METADATA_FAMILY+":"+ACCOUNT_COLUMN_NAME {
+				if !isNativeEth {
+					tokenCount++
+
+					// check if token limit is reached (requires that ETH balance is always first)
+					if tokenCount >= limit {
+						break
```

Comment:
I am not sure if it is ok to break here, as it might cause the `ret.Name` field not to be filled?

Reply:
Good point, fixed.